File: scripts/pages/notes_2.py
"""Deuxième script de notes pour Optima."""


def on_page_enter(context):
    page = context.get("page", {})
    _log(f"Bienvenue sur {page.get('name')}")


def _log(message: str) -> None:
    print(f"[OptimaScript] {message}")

File: scripts/pages/notes_2.dart
import 'package:optimascript/api.dart';

Future<void> onPageEnter(ScriptContext context) async {
  final sheet = context.api.workbook.activeSheet;
  final cell = sheet?.cellByLabel('A1');
  final pageKey = context.descriptor.key;
  cell?.setValue('Bienvenue sur la page ' + pageKey + '.');
  await context.callHost(
    'log',
    positional: <Object?>[
      'Page ' + pageKey + ' initialisée via OptimaScript Dart.',
    ],
  );
}

Future<void> onPageLeave(ScriptContext context) async {
  final pageKey = context.descriptor.key;
  await context.callHost(
    'log',
    positional: <Object?>[
      'Page ' + pageKey + ' quittée via OptimaScript Dart.',
    ],
  );
}
